version: "3.9"
name: scholar

services:
  web:
    build:
      context: .
      target: dev
    command: npm run dev
    env_file:
      - .env
    volumes:
      - .:/app
      - /app/node_modules
      - /app/.next
    ports:
      - "4246:4246"
    expose:
      - "80"

    restart: always
    depends_on:
      db:
        condition: service_healthy
    networks:
      - net

  nginx:
    build: ./nginx
    container_name: nginx
    depends_on:
      - web
    ports:
      - "80:80"
      - "443:443"
      
    volumes:
      - ./nginx:/etc/nginx/conf.d
      - /app/.next
      - /app/public
      - certbot-etc:/etc/letsencrypt
      - certbot-var:/var/lib/letsencrypt
    restart: always
    networks:
      - net

  certbot:
    image: certbot/certbot
    container_name: certbot
    volumes:
      - certbot-etc:/etc/letsencrypt
      - certbot-var:/var/lib/letsencrypt
      - web-root:/app/.next
    depends_on:
      - nginx
    command: certonly --webroot --webroot-path=/app/.next --email monashovnikolaye@gmail.com --agree-tos --no-eff-email -d home-server.tech  -d www.home-server.tech
    networks:
      - net

  db:
    container_name: db
    image: postgres:latest
    ports:
      - "5432:5432"
    env_file:
      - .env
    volumes:
      - db-data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "postgres", "-d", "scholar"]
      interval: 10s # retry after the database is ready
      timeout: 5s
      retries: 5
    networks:
      - net

volumes:
  db-data:
  certbot-etc:
  certbot-var:
  web-root:

networks:
  net:
    driver: bridge